### PR TITLE
Feature SBOL datatypes

### PR DESCRIPTION
**Pull request by guillaume-gricourt**
Hi,
The SBOL datatype is added in this PR (#15146)

SBOL format respects RDF. Serialization of SBOL v2 can be done through XML whereas SBOL v3 can be serialized with XML, turtle, ... according to [sbolstandards](https://sbolstandard.org/datamodel-specification/)

Please let me know if this implementation is suited 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
```bash
./run_tests.sh -unit test/unit/data/datatypes/test_sbol.py
```

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -569,6 +569,7 @@
     <datatype extension="ttl" type="galaxy.datatypes.triples:Turtle" display_in_upload="true"/>
     <datatype extension="rdf" type="galaxy.datatypes.triples:Rdf" display_in_upload="true"/>
     <datatype extension="jsonld" type="galaxy.datatypes.triples:Jsonld" display_in_upload="true"/>
+    <datatype extension="sbol" type="galaxy.datatypes.triples:Sbol" display_in_upload="true"/>
```
Comment (bernt-matthias) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: Can you add a description? Three is a description attribute.

Comment (guillaume-gricourt) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: Sure, done in 3e6899163ccccc7a656877fcdc04060f44fba799

File: lib/galaxy/datatypes/triples.py [hunk 2]
```diff
@@ -202,3 +204,38 @@ def set_peek(self, dataset: DatasetProtocol, **kwd) -> None:
         else:
             dataset.peek = "file does not exist"
             dataset.blurb = "file purged from disk"
+
+
+@build_sniff_from_prefix
+class Sbol(data.Text, Triples):
+    """
+    The SBOL data format
```
Comment (bernt-matthias) [inline lib/galaxy/datatypes/triples.py]: Adding a link to docs / standard would be appreciated.

Comment (guillaume-gricourt) [reply lib/galaxy/datatypes/triples.py]: Agree, added in 6ed37a2c9198256ea558e0c2010e6627c36988de

**Review by jdavcs — APPROVED** [review]
Looks good. Thank you, @guillaume-gricourt!

**Pull request merged**